utils/conda-utils.js

```diff
@@ -0,0 +1,88 @@
+/*eslint-env node*/
+/*eslint-disable no-console*/
+'use strict';
+const {spawnSync, spawn} = require('child_process'),
+    os = require('os'),
+    path = require('path'),
+    fs = require('fs'),
+    yaml = require('js-yaml'),
+    CONDA_COMMAND = 'conda';
+
+const getCondaEnvs = function () {
+    const envProcess = spawnSyncCondaProcess(['env', 'list']);
+    return envProcess.stdout.toString().split('\n')
+        .filter(line => !!line && !line.startsWith('#'))
+        .map((env) => {
+            const [name, path] = env.split(/\s+/);  //eslint-disable-line no-unused-vars
+            return name;
+        }).filter(env => !!env);
+};
+
+const envExists = function (name) {
+    const availableEnvs = getCondaEnvs();
+    return availableEnvs.indexOf(name) > -1;
+};
+
+const dumpYAML = function (environment, envFileName) {
+    if (!envFileName) {
+        envFileName = path.join(os.tmpdir(), 'deepforge.yml');
+    }
+    const envYamlString = yaml.safeDump(environment);
+    fs.writeFileSync(envFileName, envYamlString, 'utf8');
+    return envFileName;
+};
+
+const checkConda = function () {
+    const conda = spawnSyncCondaProcess(['-V']);
+    if (conda.status !== 0) {
+        throw new Error(`Please install conda before continuing. ${conda.stderr.toString()}`);
+    }
+};
+
+
+const createOrUpdateEnvironment = function (envFile) {
+    const env = yaml.safeLoad(fs.readFileSync(envFile, 'utf8'));
+    if (process.env.DEEPFORGE_CONDA_ENV && process.env.DEEPFORGE_CONDA_ENV !== env.name) {
+        env.name = process.env.DEEPFORGE_CONDA_ENV;
+        envFile = dumpYAML(env, envFile);
+    }
+    const createOrUpdate = envExists(env.name) > -1 ? 'update' : 'create';
+    console.log(`Environment ${env.name} will be ${createOrUpdate}d.`);
+    spawnCondaProcess(['env', createOrUpdate, '--file', envFile],
+        `Successfully ${createOrUpdate}ed the environment ${env.name}`);
+
+};
+
+const spawnCondaProcess = function (args, onCompleteMessage, onErrorMessage) {
+    const condaProcess = spawn(CONDA_COMMAND, args, {
+        shell: os.type === 'Windows_NT' ? true : '/bin/bash'
```

Comment:
Generally, it would be good to define a constant like `isWindows` and have the check there rather than having redundant checks. This is nice for a couple reasons:
- Easy to refactor (in case `os.platform()` is a better check for some reason)
- Minimize code duplication and related bugs. 

The second one is particularly relevant since `os.type` is not invoked so it is comparing a constant function and a constant string (always false).

Reply:
Other checks are done correctly though